src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -205,6 +205,15 @@ internal void Start()
             gcTimer = t;
         }
 
+        internal void RemoveStuckGrainFromDirectory(ActivationData activationData)
+        {
+            DeactivateActivations(new List<ActivationData>() {activationData}).Ignore();
```

Comment:
Please add comment explaining the logic here: why we call DeactivateActivation and ALSO call directory.UnregisterAsync. (DeactivateActivation  already does directory.UnregisterAsync, so we want to explain that we don't want to wait with UnregisterAsync until activation gets Unstuck).